src/Diagnostics/DiagnosticsMachine/group_gen.jl

```diff
@@ -262,8 +262,8 @@ function generate_collect_calls(
             else
                 AT2 = impl_args[2][2] # the type of the second argument
                 @assert isa_bl(AT2)
```

Comment:
I think I saw this assertion break in one of these PRs? If so, it is probably because the implementation of `isa_bl` only tests `AT2` against the immediate sub-types of `BalanceLaw`. If a deeper hierarchy of sub-types is used, it will fail. Should be an easy fix if you want to add it here, or it can be done in a separate PR.

Reply:
Let's do a separate PR (that was just for debugging something I think)